Automattic/newspack-plugin
# feat(ia): settings connections refactor various

### All Submissions:


### Changes proposed in this Pull Request:

#### Style changes:

* Added `.newspack-wizard__section` & `.newspack-wizard__sections` classes for controlling Wizard layouts comprising of lists of components.
* Required reseting a few elements to provide concise spacing.
* Opted for `rem` units for spacing to work with 8px increments.

#### Settings > Connections component:

* Moved `settings/tabs/connections/index.tsx` a level up to `settings/connections/index.tsx`.
* Added `window.newspackSettings.connections` object (and type) to house `wp_localize_script` config.
* Added boilerplate for future components.

#### Misc:
* `<SectionHeader />` got `@typedef` to appease consumption in TypeScript components.
* Removed `newspack-wizard` `className` from `<Wizard />` as it is already included in `includes/wizards/class-wizard.php`.
* Moved `wp_localize_script` data to separate method `get_local_data` in both settings and dashboard classes.
* Moved script registration for `wizards.js` to parent class `Wizard`. Avoids script registration duplication.
* Removed duplicate `<Footer />` component. 
* Added Definitely Typed packages for `lodash` and `@wordpress/api-fetch` packages.

### How to test the changes in this Pull Request:

1. Checkout this branch and build assets.
2. Navigate to /wp-admin/admin.php?page=newspack-settings#/
3. Observe that you see the below:
![np-ia local_wp-admin_admin php_page=newspack-settings (2)](https://github.com/Automattic/newspack-plugin/assets/3676975/36f8137b-eb85-480b-9caf-a50ba90d0683)

### Other information:
